app/bundles/CoreBundle/Helper/ExportHelper.php

```diff
@@ -23,6 +23,11 @@ class ExportHelper
 
     public const EXPORT_TYPE_CSV   = 'csv';
 
+    /**
+     * @var array<int, string>
+     */
+    private array $headerRow = [];
```

Comment:
I believe this class is not intended to maintain any state; it should function strictly as a service. It is injected as a dependency, and once the header row is set, the state would need to be reset if the class is to be used again to generate a second file.